tests/network_manager_test.go

```diff
@@ -78,3 +82,56 @@ func TestBlockIPs(t *testing.T) {
 	resp := p.Goto(tb.URL("/get"), nil)
 	assert.NotNil(t, resp)
 }
+
+func TestBasicAuth(t *testing.T) {
+	const (
+		validUser     = "validuser"
+		validPassword = "validpass"
+	)
+
+	browser := newTestBrowser(t, withHTTPServer()).
+		withHandler("/auth", func(w http.ResponseWriter, r *http.Request) {
+			user, pass, ok := r.BasicAuth()
+			if !ok {
+				w.Header().Set("WWW-Authenticate", `Basic realm="u=User,p=Pass"`)
+			}
+			if user != validUser || pass != validPassword {
+				http.Error(w, "Unauthorized", http.StatusUnauthorized)
+			}
```

Comment:
We don't need to implement this ourselves: [`httpbin` has a `/basic-auth/` endpoint](https://httpbin.org/#/Auth/get_basic_auth__user___passwd_). I ran your test with our `httpmultibin` and was able to reproduce the `ERR_TOO_MANY_RETRIES` issue, so we can just use that. :)

Reply:
:) Yeah, but I think that code is buggy. It does the same thing I did before—it doesn't care about `ok`.

https://github.com/mccutchen/go-httpbin/blob/b42779a256a27e37ad23b1518dc7781c3b73026f/httpbin/handlers.go#L432-L456

Reply:
Sure, but it's probably "buggy" for a reason :) Compatibility comes to mind, as I've found at least one live server that has the same behavior (I messaged you on Slack).

Regardless, I think we should handle this gracefully even for buggy servers, otherwise we risk flooding the server with useless requests.

Reply:
It seems like nginx handles it correctly (ofc, if you configured it correctly). It checks for the authorization header first and declines the request if it's not there:

https://github.com/nginx/nginx/blob/1f01183b9e6658749934313fd72f7f16c1918b54/src/http/ngx_http_core_module.c#L1958-L1961

But I can't say the same for the go-httpbin 🤷 

If you can find an underlying reason, please let me know, I'm curious!

As we discussed on slack, I'll work on handling buggy servers next week 😄

Reply:
Maybe 🤷  Let's give it a try.

Reply:
See: [c4f4923](https://github.com/grafana/xk6-browser/pull/242/commits/c4f49235ff036ca3f3b204f157a16e16b2bd4690)